tauri-macros/src/command.rs

```diff
@@ -0,0 +1,96 @@
+use proc_macro2::{Ident, TokenStream};
+use quote::{format_ident, quote};
+use syn::{parse::Parser, parse2, punctuated::Punctuated, Attribute, ItemFn, Path, Token};
+
+pub fn generate_command(_attrs: Vec<Attribute>, function: ItemFn) -> TokenStream {
+  let ident = function.sig.ident.clone();
+  let params = function.sig.inputs.clone();
+  let (mut names, mut types): (Vec<syn::Ident>, Vec<syn::Path>) = params
+    .iter()
+    .map(|param| {
+      let mut arg_name = None;
+      let mut arg_type = None;
+      match param {
+        syn::FnArg::Typed(rec) => {
+          match rec.pat.as_ref() {
+            syn::Pat::Ident(ident) => {
+              arg_name = Some(ident.ident.clone());
+            }
+            _ => (),
+          }
+          match rec.ty.as_ref() {
+            syn::Type::Path(path) => {
+              arg_type = Some(path.path.clone());
+            }
+            _ => (),
+          }
+        }
+        _ => (),
+      }
+      (arg_name.unwrap(), arg_type.unwrap())
+    })
+    .unzip();
+
+  // Default generic for functions that don't take the webview
+  let mut application_ext_generic = quote!(<A: tauri::ApplicationExt>);
+  let mut webview_arg_type = quote!(tauri::WebviewManager<A>);
+  let webview_arg = match types.first() {
+    Some(first_type) => {
+      let webview_type = parse2::<Ident>(quote!(WebviewManager).into()).unwrap();
+      if first_type.segments.last().unwrap().ident == webview_type {
+        // Use specific WebviewManager defined by the function
+        webview_arg_type = quote!(#first_type);
+        application_ext_generic = quote!();
+        types.drain(0..1);
+        names.drain(0..1);
+        quote!(_webview,)
+      } else {
+        quote!()
+      }
+    }
+    None => quote!(),
+  };
```

Comment:
Because macros have type information, this is only checking that the type ident only matches a `WebviewManager` directly.  Since that seems like an easy enough name collision that will cause unexpected (for the user) name conflicts, i would suggest detecting if there is a webview being passed into the arguments with an item.  maybe something like
```rust
#[command(webview = "webview")]
fn reply_string(webview: tauri::WebviewManager<tauri::flavors::Wry>, s: String) -> String {
  println!("{}", webview.current_window_label());
  s
}
```

It's slightly repetitive if they are both called webview, but is very explicit and can prevent conflicts.  If you know the properties of all webview managers, you can even compile-time test inside that property to spit out a more specific compile error if the `webview` item parameter isnt actually a webview type.

Reply:
Suggestion applied, thanks for the tip! However, I implemented it as a boolean that would set the first arg to be the webview.